services/worker/src/worker/main.py

```diff
@@ -123,6 +136,35 @@ def kill_zombies(self) -> None:
             Job.objects(pk__in=[zombie.pk for zombie in zombies]).update(
                 status=Status.ERROR, finished_at=get_datetime()
             )
+            for zombie in zombies:
+                job_info = JobInfo(
+                    job_id=str(zombie.pk),
+                    type=zombie.type,
+                    dataset=zombie.dataset,
+                    config=zombie.config,
+                    split=zombie.split,
+                    force=zombie.force,
+                    priority=zombie.priority,
+                )
+                job_runner = self.job_runner_factory.create_job_runner(job_info)
+                error = WorkerCrashedError("Worker crashed while running this job.")
+                upsert_response(
+                    kind=job_runner.processing_step.cache_kind,
+                    dataset=job_runner.dataset,
+                    config=job_runner.config,
+                    split=job_runner.split,
+                    content=dict(error.as_response()),
+                    http_status=error.status_code,
+                    error_code=error.code,
+                    details=dict(error.as_response_with_cause()),
+                    worker_version=job_runner.get_version(),
```

Comment:
Not sure if we should get the current version of the job, I was thinking, what if the service crashed (Included the WorkerExecutor) and we are deploying a new job version?, it will assume the version which the job crashed is the new one and not the previous one. But not sure

Reply:
Maybe we should add the job version to the `Job` objects in the queue then ? Otherwise we can't know if the zombie job is using the current version or a different one

Reply:
created https://github.com/huggingface/datasets-server/issues/830

Reply:
Related: the last dataset commit can also have changed. So: we should specify both in the job, I think.

Beware, it also means that we should add some logic to cancel previous jobs on every new commit to main.